### PR TITLE
[WIP] Pretty-print numbers ending with large number of zeros

### PR DESCRIPTION
**Pull request by bjodah**
I have been working with some expressions with quite large numbers, and I have found myself  manually counting zeros verifying the formulae. Before this pull request we have this:
```
>>> import sympy
>>> print(42*sympy.S(10)**33)
42000000000000000000000000000000000
```
after this PR the output would be:
```
42*10**33
```
I have considered str, latex and pretty printers. There seem to be a minor issue with printing parenthesis in the prettty printer. I would appreciate if someone familiar with the pretty printers could guide me to the proper fix for this.

This PR is a bit intrusive (downstream projects may have their doctests broken etc.). It is also very important that the correctness of printed values are preserved (obviously), so this should be thoroughly reviewed before merged.

### CODE REVIEW

File: sympy/printing/printer.py [hunk 1]
```diff
@@ -268,3 +271,13 @@ def _as_ordered_terms(self, expr, order=None):
             return sorted(Add.make_args(expr), key=cmp_to_key(Basic._compare_pretty))
         else:
             return expr.as_ordered_terms(order=order)
+
+
+def _print_Integer__scientific(printer, expr, limit=6, base=10, every=3):
```
Comment (asmeurer) [inline sympy/printing/printer.py]: Why is there an extra _ before scientific?

Comment (asmeurer) [inline sympy/printing/printer.py]: It would be good to add direct tests for this function.

**Comment by asmeurer** [comment]
Maybe we should add a printer option for this. There could definitely be situations where this isn't wanted.

**Comment by bjodah** [comment]
A printer option sounds like a good idea. Should the scientific notation be opt-in or opt-out (i.e. what should the default be)?

File: sympy/printing/printer.py (hunk 1 above)
Comment (bjodah) [inline sympy/printing/printer.py]: No particular reason for the double score except for indicating that it is not based on any class. Will change this. I will also add a separate test for the function.

**Comment by asmeurer** [comment]
No strong opinions either way.

**Comment by czgdp1807** [comment]
@bjodah Can we continue this PR? I think I can complete it. Please let me know of what you think.

**Comment by bjodah** [comment]
@czgdp1807 thanks! I think it would be awesome if you finished it -- I haven't found the time lately and probably won't for the foreseeable future :/
You have my blessing for sure, feel free to ping me in a new PR of yours for review or discussions.

File: sympy/polys/polytools.py [hunk 2]
```diff
@@ -6090,7 +6090,7 @@ def factor(f, *gens, **args):
     >>> factor((x**2 - 1)/(x**2 + 4*x + 4))
     (x - 1)*(x + 1)/(x + 2)**2
     >>> factor((x**2 + 4*x + 4)**10000000*(x**2 + 1))
-    (x + 2)**20000000*(x**2 + 1)
+    (x + 2)**20*10**6*(x**2 + 1)
```
Comment (bjodah) [inline sympy/polys/polytools.py]: oops. This should probably read `(x + 2)**(20*10**6)*(x**2 + 1)`

**Comment by czgdp1807** [comment]
Closing this in favour of https://github.com/sympy/sympy/pull/17853

**Pull request closed without merging**